iree/compiler/Dialect/HAL/Conversion/HALToVM/ConvertCommandBufferOps.cpp

```diff
@@ -13,6 +13,65 @@ namespace mlir {
 namespace iree_compiler {
 namespace {
 
+class CommandBufferFillBufferOpConversion
+    : public OpConversionPattern<IREE::HAL::CommandBufferFillBufferOp> {
+ public:
+  CommandBufferFillBufferOpConversion(MLIRContext *context,
+                                      SymbolTable &importSymbols,
+                                      TypeConverter &typeConverter,
+                                      StringRef importName)
+      : OpConversionPattern(typeConverter, context) {
+    importOp = importSymbols.lookup<IREE::VM::ImportOp>(importName);
+    assert(importOp);
+  }
+
+  LogicalResult matchAndRewrite(
+      IREE::HAL::CommandBufferFillBufferOp op, llvm::ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const override {
+    auto importType = importOp.getType();
+    IREE::HAL::CommandBufferFillBufferOp::Adaptor newOperands(operands);
+
+    SmallVector<Value, 8> callOperands = {
+        newOperands.command_buffer(),
+        newOperands.target_buffer(),
+        newOperands.target_offset(),
+        newOperands.length(),
+    };
+
+    // The pattern will be promoted as needed to a 32 bit type, but we still
+    // need to bitcast from float to int and track the original pattern length.
+    auto originalPatternType = op.pattern().getType();
+    auto patternBitWidth = originalPatternType.getIntOrFloatBitWidth();
+    auto patternLength = rewriter.createOrFold<mlir::arith::ConstantIntOp>(
+        op.getLoc(), patternBitWidth / 8, 32);
+    Value pattern;
+    if (originalPatternType.isInteger(8) || originalPatternType.isInteger(16) ||
+        originalPatternType.isInteger(32)) {
+      pattern = op.pattern();
+    } else if (originalPatternType.isF32()) {
+      pattern = rewriter.createOrFold<arith::BitcastOp>(
+          op.getLoc(), rewriter.getIntegerType(patternBitWidth), op.pattern());
+    } else {
+      // Note: f16 in particular would need special handling since a promotion
+      // to f32 changes the bit representation.
+      return op.emitOpError()
+             << "unhandled fill buffer type: " << originalPatternType;
+    }
+    callOperands.push_back(pattern);
+    callOperands.push_back(patternLength);
+
+    auto callOp = rewriter.replaceOpWithNewOp<IREE::VM::CallOp>(
+        op, SymbolRefAttr::get(importOp), importType.getResults(),
+        callOperands);
```

Comment:
something in here isn't handling type conversions, but I'm not sure what / how to track that down:

```
model.tflite:0:0: error: failed to materialize conversion for result #0 of operation 'arith.constant' that remained live after conversion
model.tflite:0:0: note: called from
model.tflite:0:0: note: see current operation: %139 = "arith.constant"() {value = -1 : i8} : () -> i8
model.tflite:0:0: note: see existing live user here: vm.call @hal.command_buffer.fill_buffer(%ref_94, %ref_91, %zero, %c9409, %c-1_i8, %c1_i32_96) : (!vm.ref<!hal.command_buffer>, !vm.ref<!hal.buffer>, i32, i32, i8, i32) -> ()
model.tflite:0:0: error: conversion to vm.module failed
```

any tips?